pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -768,115 +679,28 @@ private void addNewRow(int docId, GenericRow row) {
-
-          // Update forward index
-          indexContainer._forwardIndex.setDictIdMV(docId, dictIds);
-
-          // Update inverted index
-          MutableInvertedIndex invertedIndex = indexContainer._invertedIndex;
-          if (invertedIndex != null) {
-            for (int dictId : dictIds) {
-              try {
-                invertedIndex.add(dictId, docId);
-              } catch (Exception e) {
-                recordIndexingError(FieldConfig.IndexType.INVERTED, e);
-              }
-            }
-          }
-        } else {
-          // Raw MV columns
-
-          switch (dataType.getStoredType()) {
-            case INT:
-              Object[] values = (Object[]) value;
-              int[] intValues = new int[values.length];
-              for (int i = 0; i < values.length; i++) {
-                intValues[i] = (Integer) values[i];
-              }
-              indexContainer._forwardIndex.setIntMV(docId, intValues);
-              indexContainer._valuesInfo.updateMVNumValues(intValues.length);
-              break;
-            case LONG:
-              values = (Object[]) value;
-              long[] longValues = new long[values.length];
-              for (int i = 0; i < values.length; i++) {
-                longValues[i] = (Long) values[i];
-              }
-              indexContainer._forwardIndex.setLongMV(docId, longValues);
-              indexContainer._valuesInfo.updateMVNumValues(longValues.length);
-              break;
-            case FLOAT:
-              values = (Object[]) value;
-              float[] floatValues = new float[values.length];
-              for (int i = 0; i < values.length; i++) {
-                floatValues[i] = (Float) values[i];
-              }
-              indexContainer._forwardIndex.setFloatMV(docId, floatValues);
-              indexContainer._valuesInfo.updateMVNumValues(floatValues.length);
-              break;
-            case DOUBLE:
-              values = (Object[]) value;
-              double[] doubleValues = new double[values.length];
-              for (int i = 0; i < values.length; i++) {
-                doubleValues[i] = (Double) values[i];
-              }
-              indexContainer._forwardIndex.setDoubleMV(docId, doubleValues);
-              indexContainer._valuesInfo.updateMVNumValues(doubleValues.length);
-              break;
-            default:
-              throw new UnsupportedOperationException(
-                  "Unsupported data type: " + dataType + " for MV no-dictionary column: " + column);
+        Object[] values = (Object[]) value;
+        for (Map.Entry<IndexType, MutableIndex> indexEntry : indexContainer._mutableIndexes.entrySet()) {
+          try {
+            indexEntry.getValue().add(values, dictIds, docId);
+          } catch (Exception e) {
+            recordIndexingError(indexEntry.getKey(), e);
           }
         }
+        indexContainer._valuesInfo.updateMVNumValues(values.length);
```

Comment:
Here I simplify the logic. Previously we called:

```
              indexContainer._valuesInfo.updateMVNumValues(whateverValues.length);
```

where `whateverValues` was an array like `values` where each `Object` was casted to something else, but their lenght was always the same.